File: app/src/main/java/com/alex/morgan/bearlist/list/RandomBearGenerator.java
package com.alex.morgan.bearlist.list;

import com.alex.morgan.bearlist.Bear;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Random;

import javax.inject.Inject;
import javax.inject.Singleton;

import dagger.Reusable;

@Reusable
class RandomBearGenerator implements BearSource {

    private final BearNameGenerator nameGenerator;

    @Inject
    RandomBearGenerator(BearNameGenerator nameGenerator) {
        this.nameGenerator = nameGenerator;
    }

    @Override
    public Collection<Bear> getAllBears() {
        try {
            //Long operation going on here
            Thread.sleep(2000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        ArrayList<Bear> bears = new ArrayList<>();
        for (int i = 0; i < 50; i++) {
            bears.add(this.randomBear());
        }
        return bears;
    }

    private static final Random random = new Random();

    private Bear randomBear() {
        return new Bear(nameGenerator.getRandomBearName(),
                "https://placebear.com/"
                        + (random.nextInt(100) + 50) + "/"
                        + (random.nextInt(100) + 50) + ".jpg");
    }
}

File: beardata/src/main/java/com/alex/morgan/bearlist/list/RandomBearGenerator.kt
package com.alex.morgan.bearlist.list

import com.alex.morgan.bearlist.Bear
import dagger.Reusable
import java.util.ArrayList
import java.util.Random
import javax.inject.Inject

@Reusable
class RandomBearGenerator @Inject
constructor(private val nameGenerator: BearNameGenerator) : BearSource {

    override val allBears: Collection<Bear>
        get() {
            try {
                //Long operation going on here
                Thread.sleep(2000)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

            val bears = ArrayList<Bear>()
            for (i in 0..49) {
                bears.add(this.randomBear())
            }
            return bears
        }

    private fun randomBear(): Bear {
        return Bear(
            nameGenerator.randomBearName,
            "https://placebear.com/"
                    + (random.nextInt(100) + 50) + "/"
                    + (random.nextInt(100) + 50) + ".jpg"
        )
    }

    companion object {

        private val random = Random()
    }
}
